test/api/importcsv/api_test.go

```diff
@@ -0,0 +1,221 @@
+
+			// Get all imports.
+			var outputImports []api.Import
+			assert.Should(t, Client.Get(api.ImportsRoot, &outputImports))
+			j, k := 0, 0
+			for _, imp := range outputImports {
+				if imp["recordType1"] == 1 && j < len(expectedApps) {
+					// An Application with no dependencies.
+					if expectedApps[j].Name != imp["applicationName"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedApps[j].Name, imp["applicationName"])
+					}
+					if expectedApps[j].Description != imp["description"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedApps[j].Description, imp["description"])
+					}
+					if expectedApps[j].BusinessService.Name != imp["businessService"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedApps[j].BusinessService.Name, imp["businessService"])
+					}
+					j++
+				}
+				if imp["recordType1"] == 2 && k < len(expectedDeps) {
+					// An Application with Dependencies.
+					if expectedDeps[k].From.Name != imp["applicationName"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedDeps[k].From.Name, imp["applicationName"])
+					}
+					if expectedDeps[k].To.Name != imp["dependency"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedDeps[k].To.Name, imp["dependency"])
+					}
+					k++
+				}
+			}
+
+			// Download the csv.
+			pathToOutputCSV := "downloadcsv.csv"
+			err := Client.FileGet(api.DownloadRoot, pathToOutputCSV)
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			// Compare contents of the csv.
+			file1, err := os.Open(pathToOutputCSV)
+			if err != nil {
+				t.Errorf(err.Error())
+				return
+			}
+			defer file1.Close()
+
+			// Open the second CSV file
+			file2, err := os.Open(r.FileName)
+			if err != nil {
+				t.Errorf(err.Error())
+				return
+			}
+			defer file2.Close()
+
+			// Read both the CSV files for comparison.
+			reader1 := csv.NewReader(file1)
+			reader2 := csv.NewReader(file2)
+
+			column1, err := reader1.Read()
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			columm2, err := reader2.Read()
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			// Check number of columns.
+			if len(column1) != len(columm2) {
+				t.Errorf("The Content of both the CSV files are different")
+			}
+
+			// Check column names.
+			for i := range column1 {
+				if column1[i] != columm2[i] {
+					t.Errorf("Mismatch in the Column Names")
+				}
+			}
+
+			// Compare rest of the contents of the files.
+			reader1.FieldsPerRecord = -1
+			item1, err := reader1.ReadAll()
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			reader2.FieldsPerRecord = -1
+			item2, err := reader2.ReadAll()
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			// Compare number of records present.
+			if len(item1) != len(item2) {
+				t.Errorf("Mismatch in number of records present")
+			}
+
+			// Compare each value.
+			for i := 0; i < len(item1); i++ {
```

Comment:
Happy to see such precise file content diff check (byte by byte), but it might be easier/more matching to the test/better to read both files data, cast []byte to string and compare it and in case of error, print both contents got and expected, that will be much easier to debug why a test potentialy failed.